### PR TITLE
Api runthrough

### LINKED ISSUE #213: Run through current API and add error handling to "silently failing" calls + some refactoring
**Describe the solution you'd like**
Do a clean run through the currently implemented API and update any error handling, and other things.

casusController:
* Add default case to the switch

marketController:
* Actually `return`ing responses down the line

replayController:
* Remove unused dependencies
* Actually `return`ing responses down the line

tankController:
* Add error validation to some DB calls in casusUpdate, tankUpdate, assignTank
* Actually `return`ing responses down the line

userController:
* Add error validation to DB calls down the line
* Remove unused dependencies
* `return`ing responses down the line

auth;
* `return`ing responses down the line

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Derplime**
**Description**
Did a run-through of the APIs to clean them up.
Any additions were for error handling or consistency of returning responses where necessary.
Overall none of these updates affects functionality.
See #213 for a general list of what I ran through. Not everything is there, but it's a example list of what I looked into.

Resolves #213 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
* Verify any desired API still works after these changes.

Passes flow:
image

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/backend/controllers/userController.js [hunk 1]
```diff
@@ -1,24 +1,20 @@
 // @flow strict
 
 // Required imports
+import type { $Request,	$Response } from 'express';
```
Comment (Derplime) [inline src/backend/controllers/userController.js]: Not sure why it looks weird here, it looks normal when you go to edit it here and on my local machine

File: src/backend/controllers/tankController.js [hunk 2]
```diff
@@ -1,64 +1,67 @@
...
-} from 'express';
-
-exports.getFavorite = async (req: $Request, res: $Response) => {
+exports.getFavorite = async (req: Request, res: Response) => {
 	await User.findById(req.user.id, 'favoriteTankId', (err: Error, myUser: User) => {
 		if(err) {
 			res.send(err);
 			console.log('could not find user');
 		} 
 		else
```
Comment (SecondThread) [inline src/backend/controllers/tankController.js]: err, I think you are missing curly braces on your else if you want the retrieved... to only be in the else?

File: src/backend/controllers/casusController.js [hunk 3]
```diff
@@ -1,23 +1,22 @@
 // @flow strict
 
-const User = require('../../models/userModel');
+// NOTE: This functionality is not implemented in v1.0
```
Comment (SecondThread) [inline src/backend/controllers/casusController.js:3]: Someday you will be formally declared a saint, Emil, for all your helpful documentation to future groups.

File: src/backend/controllers/tankController.js [hunk 4]
```diff
@@ -1,64 +1,67 @@
...
 		}
 		else{
 			res.send(result.favoriteTankId);
+			console.log('Updated favorite tank.');
 		}
 	});
 }
 
-exports.userTanks = async (req: $Request, res: $Response) => {
+exports.userTanks = async (req: Request, res: Response) => {
```
Comment (SecondThread) [inline src/backend/controllers/tankController.js:38]: thanks for fixing

File: src/backend/controllers/userController.js [hunk 5]
```diff
@@ -1,24 +1,20 @@
...
-const crypto = require('crypto');
-const nodemailer = require('nodemailer');
-import type {
-	$Request,
-	$Response,
-	NextFunction,
-	Middleware,
-} from 'express';
+// Throws an error if this isn't here because of async functions
+const regeneratorRuntime = require("regenerator-runtime");
```
Comment (SecondThread) [inline src/backend/controllers/userController.js]: is this used?

File: src/backend/routes/casusRoutes.js [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
 
-//==========================================================================//
-// TO USE ROUTES: all route calls in this file will be /api/cassus/<Route call> //
-//==========================================================================//
+//=============================================================================//
+// TO USE ROUTES: all route calls in this file will be /api/casus/<Route call> //
+//=============================================================================//
 
-// Update the amount of cassus blocks a player has
+// Update the amount of casus blocks a player has
```
Comment (SecondThread) [inline src/backend/routes/casusRoutes.js:16]: <3

File: src/backend/controllers/userController.js (hunk 1 above)
Comment (SecondThread) [reply src/backend/controllers/userController.js]: you had a tab instead of a space

File: src/backend/controllers/userController.js (hunk 5 above)
Comment (Derplime) [reply src/backend/controllers/userController.js]: Yeah I haven't really figured out why only this controller requires it while the others don't.

Comment (SecondThread) [reply src/backend/controllers/userController.js]: hmm. Odd. Well if you never use the variable, you can change the line to just:
`require("regenerator-runtime");`

**Review by SecondThread — APPROVED** [review]
Just fix the curly braces on the else, but other than that lgtm

**Pull request merged**